src/cli/cli.cpp

```diff
@@ -139,10 +141,14 @@ Interpreter::Interpreter(Instance *aInstance, otCliOutputCallback aCallback, voi
 #if OPENTHREAD_FTD
     otThreadSetDiscoveryRequestCallback(mInstance, &Interpreter::HandleDiscoveryRequest, this);
 #endif
+
+    OutputPrompt();
 }
 
 void Interpreter::OutputResult(otError aError)
 {
+    OT_ASSERT(mCommandIsPending);
+
```

Comment:
Current code is correct but just a suggestion to maybe make it simpler: 
How about if we do a `VerifyOrExit(aError != OT_ERROR_PENDING)` at the top here?
Then we can remove both `case OT_ERROR_PENDING` and also the `if (aError != OT_ERROR_PENDING)` check.